guide/english/certifications/apis-and-microservices/apis-and-microservices-projects/timestamp-microservice/index.md

```diff
@@ -1,10 +1,91 @@
+
+**Solution ahead!**
+
+## ![:beginner:](https://forum.freecodecamp.com/images/emoji/emoji_one/beginner.png?v=3 ':beginner:') Basic Code Solution:
+
+```javascript
+app.get('/api/timestamp/', (req, res) => {
+  res.json({ unix: Date.now(), utc: Date() });
+});
+
+app.get('/api/timestamp/:date_string', (req, res) => {
+  let dateString = req.params.date_string;
+
+  //A 4 digit number is a valid ISO-8601 for the beginning of that year
+  //5 digits or more must be a unix time, until we reach a year 10,000 problem
+  if (/\d{5,}/.test(dateString)) {
+    dateInt = parseInt(dateString);
+    //Date regards numbers as unix timestamps, strings are processed differently
+    res.json({ unix: dateString, utc: new Date(dateInt).toUTCString() });
+  }
+
+  let dateObject = new Date(dateString);
+
+  if (dateObject.toString() === 'Invalid Date') {
+    res.json({ error: 'Invaid Date' });
+  } else {
+    res.json({ unix: dateObject.valueOf(), utc: dateObject.toUTCString() });
+  }
+});
+```
+
+### Code Explanation:
+
+- This is a pretty straightforward application of the lessons, Basic Node and Express - Serve JSON on a Specific Route
+  and Basic Node and Express - Get Route Parameter Input from the Client. The added wrinkle is the use of Javascript's native Date object.
+- The route parameter is assigned to the variable dateString, and processed accordingly. If it is a 5-or more digit number, it is assumed to be a UNIX timestamp format, and cast into a number and passed to a Date object. Otherwise, it is passed to the Date object as a string.
+- This code has sequential if and if-else statements instead of a switch statement because it is easier to rely on the Date object's ability to reject invalid date formats.
+
+That's it. There is no intermediate or advanced solution yet. If you have a better, more elegant solution, help us all out and contribute them!
+
+## ![:clipboard:](https://forum.freecodecamp.com/images/emoji/emoji_one/clipboard.png?v=3 ':clipboard:') NOTE TO CONTRIBUTORS:
 
-<a href='https://github.com/freecodecamp/guides/blob/master/README.md' target='_blank' rel='nofollow'>This quick style guide will help ensure your pull request gets accepted</a>.
+- ![:warning:](https://forum.freecodecamp.com/images/emoji/emoji_one/warning.png?v=3 ':warning:') **DO NOT** add solutions that are similar to any existing solutions. If you think it is **_similar but better_**, then try to merge (or replace) the existing similar solution.
+- Add an explanation of your solution.
+- Categorize the solution in one of the following categories -- **Basic**, **Intermediate** and **Advanced**. ![:traffic_light:](https://forum.freecodecamp.com/images/emoji/emoji_one/traffic_light.png?v=3 ':traffic_light:')
+- Please add your username only if you have added any **relevant main contents**. (![:warning:](https://forum.freecodecamp.com/images/emoji/emoji_one/warning.png?v=3 ':warning:') **_DO NOT_** _remove any existing usernames_)
```

Comment:
Please remove this line as this is no longer done at FCC.
The names of contributors is not added because it is tough to categorise what content is for the lack of a better word - `main` or not.

Reply:
Let's merge this one then @RandellDawson